Dockerfile.hpu.ubi

```diff
@@ -84,18 +96,42 @@ RUN --mount=type=bind,from=build,src=/workspace/dist,target=/workspace/dist \
     pip install $(echo dist/*.whl)'[tensorizer]' --verbose
 
 ENV HF_HUB_OFFLINE=1 \
-    PORT=8000 \
     HOME=/home/vllm \
-    VLLM_USAGE_SOURCE=production-docker-image
+    VLLM_USAGE_SOURCE=production-docker-image \
+    VLLM_NO_USAGE_STATS=1
 
 # setup non-root user for OpenShift
-# In OpenShift the user ID is randomly assigned, for compatibility we also
-# set up a non-root user here.
-RUN umask 002 \
-    && useradd --uid 2000 --gid 0 vllm \
-    && chmod g+rwx $HOME /usr/src /workspace
+RUN umask 002 && \
+    useradd --uid 2000 --gid 0 vllm && \
+    mkdir -p /home/vllm && \
+    chown -R vllm /home/vllm && \
+    chmod g+rwx /home/vllm
 
 COPY LICENSE /licenses/vllm.md
+COPY examples/*.jinja /app/data/template/
 
 USER 2000
+WORKDIR /home/vllm
+
 ENTRYPOINT ["python3", "-m", "vllm.entrypoints.openai.api_server"]
+
+## vllm-grpc-adapter #####################################################################
+FROM vllm-openai as vllm-grpc-adapter
+
+USER root
+
+ARG VLLM_TGIS_ADAPTER_VERSION
+RUN --mount=type=cache,target=/root/.cache/pip \
+    --mount=type=bind,from=build,src=/workspace/dist,target=/workspace/dist \
+    pip install vllm-tgis-adapter==${VLLM_TGIS_ADAPTER_VERSION}
```

Comment:
I recommend also installing the vllm wheel at the same time: if the vllm-tgis-adapter version is incompatible with vllm, the installed vllm version will get uninstalled and the upstream version (from PyPI) will be installed.